contracts/ethereum_reorg_util.go

```diff
@@ -27,28 +27,36 @@ func AwaitMining(c client.BlockchainClient) error {
 
 // NextBlockConfirmer await for the next block
 type NextBlockConfirmer struct {
-	doneChan chan struct{}
-	done     bool
-	ctx      context.Context
-	cancel   context.CancelFunc
+	doneChan  chan struct{}
+	done      bool
+	extraData []byte
+	ctx       context.Context
+	cancel    context.CancelFunc
 }
 
 // NewNextBlockConfirmer generic next block confirmer
-func NewNextBlockConfirmer() *NextBlockConfirmer {
+func NewNextBlockConfirmer(extraData []byte) *NextBlockConfirmer {
 	ctx, cancel := context.WithTimeout(context.Background(), DAGAwaitTimeout)
 	return &NextBlockConfirmer{
-		done:     false,
-		doneChan: make(chan struct{}),
-		ctx:      ctx,
-		cancel:   cancel,
+		done:      false,
+		doneChan:  make(chan struct{}),
+		extraData: extraData,
+		ctx:       ctx,
+		cancel:    cancel,
 	}
 }
 
-func (f *NextBlockConfirmer) ReceiveBlock(_ *types.Block) error {
+func (f *NextBlockConfirmer) ReceiveBlock(b *types.Block) error {
 	if f.done {
 		return nil
 	}
-	log.Info().Msg("First block received")
+	// in case of a multi-node setup we may need a block from the node on which reorg will be performed
+	if f.extraData != nil {
```

Comment:
Must wait for tx node to mine its own block, so the network is joined before splitting.